backend/scripts/calculateAirdrop/getInitialUserSnapshot.ts

```diff
@@ -0,0 +1,102 @@
+import { task } from "hardhat/config";
+import fs from "fs";
+import path from "path";
+import { allPairDataSwapWithoutGauge, getUsers } from "../../subgraph/fetchers";
+import PAIR_ABI from "../../constants/abis/pairABI.json";
+import GAUGE_ABI from "../../constants/abis/gaugeABI.json";
+import { multicallSplitOnOverflow } from "../../lib/multicall";
+import { BigNumber, ethers } from "ethers";
+import { Pair, UserData } from ".";
+
+//used to retrieve all users liq+gauge balances @ the given blocknumber
+async function getBalanceData(
+  users: string[],
+  pair: Pair,
+  blocknumber: string,
+  provider: ethers.providers.JsonRpcProvider
+): Promise<BigNumber[]> {
+  const calls = users.map((id: string) => ({
+    methodName: "balanceOf(address)",
+    methodParameters: [id],
+  }));
+  let results = await multicallSplitOnOverflow(
+    pair.address,
+    PAIR_ABI,
+    calls,
+    "SwapPair",
+    provider,
+    {
+      maxCallsPerBatch: 300,
+      blockNumber: blocknumber,
+    }
+  );
+  let balance = results.map((result) => BigNumber.from(result[0].hex));
+  if (pair.gaugeAddress) {
+    const calls = users.map((id: string) => ({
+      methodName: "balanceOf(address)",
+      methodParameters: [id],
+    }));
+    const gaugeResults = await multicallSplitOnOverflow(
+      pair.gaugeAddress,
+      GAUGE_ABI,
+      calls,
+      "Gauge",
+      provider,
+      {
+        maxCallsPerBatch: 300,
+        blockNumber: blocknumber,
+      }
+    );
+    balance = gaugeResults.map((result, index) =>
+      BigNumber.from(result[0].hex).add(balance[index])
+    );
+  }
+  return balance;
+}
+
+task(
+  "get-all-users-initial",
+  "Initial LP balances taken at the start of the period, this is used as the reference"
+)
+  .addParam("blocknumber", "Block Number")
+  .addVariadicPositionalParam("addressesArray")
+  .setAction(async ({ blocknumber, addressesArray }, { network, ethers }) => {
+    let userBalancesForPool: Record<string, UserData> = {};
+    const swapPairs = await allPairDataSwapWithoutGauge(network.name);
```

Comment:
Just looking at the name of the function, it seems that we are not fetching the Gauge used in the helper method above (`getBalanceData`). Are we sure that we are getting a gauge if present?

This code can silently introduce a wrong behavior due to the if condition at line 34.

Reply:
Just looking at the name of the function, it seems that we are not fetching the Gauge used in the helper method above (`getBalanceData`). Are we sure that we are getting a gauge if present?

This code can silently introduce a wrong behavior due to the if condition at line 34.

Reply:
Checked the query and we are fetching the `gaugeAddress` so probably just a naming issue.

Reply:
yessir, naming is a bit over the place. Ill fix theat